ablator/main/state.py

```diff
@@ -548,7 +548,7 @@ def _init_trials(self, resume: bool = False) -> list[ParallelConfig]:
         ]:
             trials_to_sample = self.n_trials_remaining
         else:
-            trials_to_sample = max_trials_conc
+            trials_to_sample = min(max_trials_conc, self.n_trials_remaining)
```

Comment:
Assume someone has a config: total_trials=20, concurrent_trials=20, and ablator runs from resume with 3 compeleted trials, so if we just take max_trials_conc, it will still sample 20 trials and the total is 23, which is not expected.

Reply:
great! Can we write a test case for this please